pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -165,17 +165,7 @@ func convertWorkloadInstances(wle *networking.WorkloadEntry, services []*model.S
 	out := make([]*model.ServiceInstance, 0)
 	for _, service := range services {
 		for _, port := range se.Ports {
-
-			// * Use security.istio.io/tlsMode if its present
-			// * If not, set TLS mode if ServiceAccount is specified
-			tlsMode := model.DisabledTLSModeLabel
-			if val, exists := wle.Labels[model.TLSModeLabelName]; exists {
-				tlsMode = val
-			} else if wle.ServiceAccount != "" {
-				tlsMode = model.IstioMutualTLSModeLabel
-			}
-			ep := convertEndpoint(service, port, wle, tlsMode)
-			out = append(out, ep)
+			out = append(out, convertEndpoint(service, port, wle))
```

Comment:
Why are you changing this? You are changing semantics here, we want tls if SA is set

Reply:
That means if sa specified in WorkloadEntry, it is mtls. 

Then why pod always has sa, it is not that. It need to provide a label

Reply:
a pod doesn't provide the label though, we automatically inject it

Reply:
I know that, but when we convert pod to WorkloadEntry it will bring confusion.

Reply:
It is what #23216 is doing. Don't we want consistency?

Reply:
do we plan to convert pod?

Reply:
I am no longer converting pod to workload Entry. Instead, using the instance handers and processing ServiceInstances (that already has the endpoint converted from Kubernetes).

Reply:
I see that, I will revert it. Just a diff here